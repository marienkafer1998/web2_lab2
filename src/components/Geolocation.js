import React from "react";
import { connect } from "react-redux";

import { setGeolocation, fetchGeoError } from "../redux/actions/geoAction";

class Geolocation extends React.Component {
    componentDidMount() {
        this.getGeolocation();
    }

    render() {
        return (
            <div >
                <button
                        onClick={() => this.getGeolocation()}
                >Get geolocation</button>
                {this.props.coords && <p>{this.props.coords.lon}, {this.props.coords.lat}</p>}
            </div>
        );
    }


    getGeolocation() {
        if (navigator.geolocation) {
            navigator.geolocation.getCurrentPosition(position => {
                    const coords = {
                        lon: position.coords.longitude,
                        lat: position.coords.latitude

                    };
                    this.props.setGeolocation(coords);
                },
                () => {
                    this.props.setGeolocation({lon: 37.62, lat: 55.75});
                });
        }
    }
}


function mapStateToProps(state) {
    return {
        coords: state.geo.coords
    };
}

function mapDispatchToProps(dispatch) {
    return {
        setGeolocation: (coords) => {
            dispatch(setGeolocation(coords));
        }
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Geolocation);