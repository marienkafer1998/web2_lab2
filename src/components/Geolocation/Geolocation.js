import React from "react";
import {connect} from "react-redux";
import MainWeather from "../MainWeather/MainWeather";
import {setGeolocation, loadWeatherWithCoords} from "../../redux/actions";
import Loader from "../Loader/Loader";
import {Button, Container, Col, Row} from "reactstrap";




class Geolocation extends React.Component {

    componentDidMount() {
        this.getGeolocation();
    }

    render() {
        console.log('rendering', this.props);
        let body =
            <Container>
                {/*<Col>*/}
                {/*Weather is here*/}
                {/*</Col>*/}
                <Col>
                    <Button color="secondary" size="lg"
                        onClick={() => this.getGeolocation()}>Get geolocation
                    </Button>
                    {this.props.response && this.props.coords &&
                    <MainWeather
                        onFetch={() => this.props.loadWeatherWithCoords(this.props.coords)}
                        weather={this.props.response}/>
                    }
                    {!this.props.response && <Loader />}

                    {!this.props.coords && <div>Error: there is no geolocation</div>}
                </Col>
            </Container>

        return (
            <Container className="weather">
                {body}
            </Container>
        );
    }


    getGeolocation() {
        console.log('in getGeoloc ' + this.props.response);
        if (navigator.geolocation) {
            navigator.geolocation.getCurrentPosition(position => {
                    const coords = {
                        lon: position.coords.longitude.toFixed(2),
                        lat: position.coords.latitude.toFixed(2)
                    };
                    this.props.setGeolocation(coords);
                    this.props.loadWeatherWithCoords(coords);
                    console.log('after getting ' + this.props.response);
                },
                () => {
                    const coords = {
                        lon: 37.62,
                        lat: 55.75

                    }
                    this.props.setGeolocation(coords);
                    this.props.loadWeatherWithCoords(this.props.coords);
                    console.log('after getting ' + this.props.response);

                });
        }
    }
}


function mapStateToProps(state) {
    return {
        coords: state.geo.coords,
        response: state.geo.response
    };
}



function mapDispatchToProps(dispatch) {
    return {
        setGeolocation: (coords) => {
            dispatch(setGeolocation(coords));
        },

        loadWeatherWithCoords: (coords) => {
            dispatch(loadWeatherWithCoords(coords));
        }
    };
}


export default connect(mapStateToProps, mapDispatchToProps)(Geolocation);