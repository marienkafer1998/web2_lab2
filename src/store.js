import { createStore, combineReducers, applyMiddleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { LOCAL_STORAGE_KEY } from "./LocalStorage";

import geoReducer from "./redux/geoReducer";
import favReducer from "./redux/favReducer";

const store = createStore(
    combineReducers({geo: geoReducer, fav: favReducer}),
    applyMiddleware(logger, thunk)
);



store.subscribe(() => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify([...store.getState().fav.favorites.keys()]));
});

export default store;
