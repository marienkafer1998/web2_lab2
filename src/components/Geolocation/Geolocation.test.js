import React from 'react';
import renderer from 'react-test-renderer';
import Geolocation from "./Geolocation.js";
import {Provider} from "react-redux";
import store from "../../store";

it('Geolocation functional rendered properly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <Geolocation/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

