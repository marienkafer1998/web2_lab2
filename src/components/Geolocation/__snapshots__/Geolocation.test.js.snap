// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geolocation functional rendered properly 1`] = `
<div
  className="weather container"
>
  <div
    className="container"
  >
    <div
      className="col"
    >
      Weather is here
    </div>
    <div
      className="col"
    >
      <button
        aria-label={null}
        className="btn btn-secondary btn-lg"
        onClick={[Function]}
        type="button"
      >
        Get geolocation
      </button>
      <div
        className="container"
      >
        <h1>
          Please, wait, weather is loading...
        </h1>
        <img
          alt="Weather is loading"
          src="https://anatomised.com/wp-content/uploads/2016/05/spinner-test4.gif"
          width="100px"
        />
      </div>
      <div>
        Error: there is no geolocation
      </div>
    </div>
  </div>
</div>
`;

exports[`Geolocation functional rendered properly should has city forecast 1`] = `
<div
  className="weather container"
>
  <div
    className="container"
  >
    <div
      className="col"
    >
      <button
        aria-label={null}
        className="btn btn-secondary btn-lg"
        onClick={[Function]}
        type="button"
      >
        Get geolocation
      </button>
      <div
        className="container"
      >
        <h1>
          Please, wait, weather is loading...
        </h1>
        <img
          alt="Weather is loading"
          src="https://anatomised.com/wp-content/uploads/2016/05/spinner-test4.gif"
          width="100px"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherParam functional rendered properly 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;

exports[`WeatherParameter functional rendered properly 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;
